src/Dapr.Client/DaprClientBuilder.cs

```diff
@@ -16,84 +17,119 @@ namespace Dapr.Client
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DaprClientBuilder"/> class.
+        /// </summary>
+        public DaprClientBuilder()
+        {
+            var daprGrpcPort = Environment.GetEnvironmentVariable("DAPR_GRPC_PORT") ?? DefaultDaprGrpcPort;
+            this.DaprGrpcEndpoint = $"http://127.0.0.1:{daprGrpcPort}";
+
+            var daprHttpPort = Environment.GetEnvironmentVariable("DAPR_HTTP_PORT") ?? DefaultDaprHttpPort;
+            this.DaprHttpEndpoint = $"http://127.0.0.1:{daprHttpPort}";
+
+            this.GrpcChannelOptions = new GrpcChannelOptions()
+            { 
+                // The gRPC client doesn't throw the right exception for cancellation
+                // by default, this switches that behavior on.
+                ThrowOperationCanceledOnCancellation = true,
+            };
+
+            this.JsonSerializerOptions = new JsonSerializerOptions(JsonSerializerDefaults.Web);
+        }
 
         // property exposed for testing purposes
-        internal string DaprEndpoint => this.daprEndpoint;
+        internal string DaprGrpcEndpoint { get; private set; }
 
         // property exposed for testing purposes
-        internal JsonSerializerOptions JsonSerializerOptions => this.jsonSerializerOptions;
+        internal string DaprHttpEndpoint { get; private set; }
 
         // property exposed for testing purposes
-        internal GrpcChannelOptions GRPCChannelOptions => this.gRPCChannelOptions;
+        internal JsonSerializerOptions JsonSerializerOptions { get; private set; }
 
+        // property exposed for testing purposes
+        internal GrpcChannelOptions GrpcChannelOptions { get; private set; }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="DaprClientBuilder"/> class.
+        /// Overrides the HTTP endpoint used by <see cref="DaprClient" /> for communicating with the Dapr runtime.
         /// </summary>
-        public DaprClientBuilder()
+        /// <param name="daprEndpoint">
+        /// The URI endpoint to use for HTTP calls to the Dapr runtime. The default value will be 
+        /// <c>http://127.0.0.1:DAPR_HTTP_PORT</c> where <c>DAPR_HTTP_PORT</c> represents the value of the 
+        /// <c>DAPR_HTTP_PORT</c> environment variable.
+        /// </param>
+        /// <returns>The <see cref="DaprClientBuilder" /> instance.</returns>
+        public DaprClientBuilder UseHttpEndpoint(string daprEndpoint)
         {
-            var daprGrpcPort = Environment.GetEnvironmentVariable("DAPR_GRPC_PORT") ?? DefaultDaprGrpcPort;
-            this.daprEndpoint = $"http://127.0.0.1:{daprGrpcPort}";
-            this.jsonSerializerOptions = new JsonSerializerOptions
-            {
-                PropertyNameCaseInsensitive = true
-            };
+            ArgumentVerifier.ThrowIfNullOrEmpty(daprEndpoint, nameof(daprEndpoint));
+            this.DaprGrpcEndpoint = daprEndpoint;
```

Comment:
this. DaprHttpEndpoint instead of grpc

Reply:
add UT

Reply:
Adding validation in build where we convert to URIs

Reply:
> add UT 

For a method that sets a property? Not worth the 3 minutes it would take.

Reply:
I'll add tests for the new validation.